lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -243,6 +243,12 @@ func (c *Client) start() {
 		// calls handle_bitmap repeatedly with the incoming bitmaps.
 		if err := cgoError(C.read_rdp_output(c.rustClient, C.uintptr_t(h))); err != nil {
 			c.cfg.Log.Warningf("Failed reading RDP output frame: %v", err)
+
+			// close the TDP connection to the browser
+			// (without this the input streaming goroutine will hang
+			// waiting for user input)
+			c.cfg.Conn.SendError("There was an error reading data from the Windows Desktop")
```

Comment:
Any chance we can cover this condition with a test?

Reply:
I'll have to think about it. I only discovered this corner case when I intentionally (and temporarily) modified the Rust to fail in a spot where it was previously impossible to fail.

Reply:
Yeah, covering this with a Go test would require building and linking to a Rust library that simulates the failure. Would be a heavy lift, so I think I'll put this in the bucket of "testing cgo is really hard."

Reply:
Fair enough.